sci-misc/boinc/boinc-7.24.1.ebuild

```diff
@@ -0,0 +1,191 @@
+HOMEPAGE="https://boinc.berkeley.edu/"
+
+SRC_URI="X? ( https://boinc.berkeley.edu/logo/boinc_glossy2_512_F.tif -> ${PN}.tif )"
+if [[ ${PV} == *9999 ]] ; then
+	EGIT_REPO_URI="https://github.com/BOINC/${PN}.git"
+	inherit git-r3
+else
+	SRC_URI+=" https://github.com/BOINC/boinc/archive/client_release/${MY_PV}/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="amd64 arm64 ~ia64 ~ppc ppc64 sparc ~x86"
+	S="${WORKDIR}/${PN}-client_release-${MY_PV}-${PV}"
+fi
+
+LICENSE="LGPL-3"
+SLOT="0"
+
+IUSE="X cuda curl_ssl_gnutls +curl_ssl_openssl opencl"
+
+REQUIRED_USE="
+	^^ ( curl_ssl_gnutls curl_ssl_openssl )
+"
+
+# libcurl must not be using an ssl backend boinc does not support.
+# If the libcurl ssl backend changes, boinc should be recompiled.
+DEPEND="
+	acct-user/boinc
+	app-misc/ca-certificates
+	cuda? (
+		x11-drivers/nvidia-drivers
+	)
+	opencl? (
+		virtual/opencl
+	)
+	dev-libs/openssl:=
+	net-misc/curl[curl_ssl_gnutls(-)=,-curl_ssl_nss(-),curl_ssl_openssl(-)=,-curl_ssl_axtls(-),-curl_ssl_cyassl(-)]
+	sys-libs/zlib
+	X? (
+		dev-libs/glib:2
+		media-libs/freeglut
+		media-libs/libjpeg-turbo:=
+		x11-libs/gtk+:3
+		x11-libs/libnotify
+		x11-libs/libX11
+		x11-libs/libXScrnSaver
+		x11-libs/libxcb:=
+		x11-libs/wxGTK:${WX_GTK_VER}[X,opengl,webkit]
+		x11-libs/xcb-util
+	)
+"
+BDEPEND="app-text/docbook-xml-dtd:4.4
+	app-text/docbook2X
+	sys-devel/gettext
+	X? ( virtual/imagemagick-tools[png,tiff] )
+"
+RDEPEND="
+	${DEPEND}
+	sys-apps/util-linux
+	!app-admin/quickswitch
+"
+
+pkg_setup() {
+	# Bug 578750
+	if use kernel_linux; then
+		linux-info_pkg_setup
+		if ! linux_config_exists; then
+			ewarn "Can't check the linux kernel configuration."
+			ewarn "You might be missing vsyscall support."
+		elif kernel_is -ge 4 4 \
+			&& linux_chkconfig_present LEGACY_VSYSCALL_NONE; then
+			ewarn "You do not have vsyscall emulation enabled."
+			ewarn "This will prevent some boinc projects from running."
+			ewarn "Please enable vsyscall emulation:"
+			ewarn "    CONFIG_LEGACY_VSYSCALL_EMULATE=y"
+			ewarn "in /usr/src/linux/.config, to be found at"
+			ewarn "    Processor type and features --->"
+			ewarn "        vsyscall table for legacy applications (None) --->"
+			ewarn "            (X) Emulate"
+			ewarn "Alternatively, you can enable CONFIG_LEGACY_VSYSCALL_NATIVE."
+			ewarn "However, this has security implications and is not recommended."
+		fi
+	fi
+}
+
+src_prepare() {
+	default
+
+	# prevent bad changes in compile flags, bug 286701
+	sed -i -e "s:BOINC_SET_COMPILE_FLAGS::" configure.ac || die "sed failed"
+
+	eautoreconf
+}
+
+src_configure() {
+	use X && setup-wxwidgets
+
+	append-libs -L"${ESYSROOT}"/usr/$(get_libdir) -L"${ESYSROOT}"/$(get_libdir)
+
+	econf --disable-server \
+		--enable-client \
+		--enable-dynamic-client-linkage \
+		--disable-static \
+		--enable-unicode \
+		--with-ssl \
+		$(use_with X x) \
+		$(use_enable X manager) \
+		$(usex X --with-wx-config="${WX_CONFIG}" --without-wxdir)
```

Comment:
I suggest the following here: (an array is safer in case a `\` would be forgotten in the future)
```
local myeconfargs=(
	--disable-server
	--enable-client
	--enable-dynamic-client-linkage
	--disable-static
	--enable-unicode
	--with-ssl
	$(use_with X x)
	$(use_enable X manager)
	$(usex X --with-wx-config="${WX_CONFIG}" --without-wxdir
)
econf "${myeconfargs[@]}"
```

Reply:
You are right, I like the idea.

Reply:
done